limiter/limiter.go

```diff
@@ -24,11 +25,15 @@ func New(maxRequests, rps int, maxDuration time.Duration) *Run {
 	}
 
 	r := &Run{
+		unlimited:    *atomic.NewBool(maxRequests == 0),
```

Comment:
Just because the reads and writes race doesn't mean it needs to be atomic. It means that you can get the wrong read value while a write is happening.

If this is non-atomic it means that in a race scenario you could do an extra request which is not a problem afaik.